src/Sylius/Component/Core/Model/ProductVariant.php

```diff
@@ -444,4 +445,12 @@ public function removeImage(ProductImageInterface $image): void
             $this->images->removeElement($image);
         }
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function compareTo($other): int
+    {
+        return $this->code === $other->getCode() ? 0 : 1;
```

Comment:
> * This method should not check for identity using ===, only for semantical equality for example
> * when two different DateTime instances point to the exact same Date + **TZ.**
> https://github.com/doctrine/common/blob/master/lib/Doctrine/Common/Comparable.php#L20-L21

I would also like to suggest using another interface because this 0 for :milk_glass:  and 1 for :coffee:  is really tricky.

Reply:
If I understood correctly, for us it is indeed semantical equality when the codes are identical. Strong cooperation is used here more because of our standards rather than real need (as both codes will be strings)